src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -269,6 +278,30 @@ private static Uri CreateConnectUrl(Uri url, NegotiationResponse negotiationResp
             }
         }
 
+        private TransferMode GetTransferMode()
+        {
+            var transferModeFeature = Features.Get<ITransferModeFeature>();
+            if (transferModeFeature == null)
+            {
+                // REVIEW: use TransferMode.Text as default if the feature is missing?
+                throw new InvalidOperationException("TransferModeFeature not set.");
+            }
+
+            return transferModeFeature.TransferMode;
+        }
+
+        private void SetTransferMode(TransferMode transferMode)
```

Comment:
Oh that. This code cannot be reached at the moment. We first call GetTransferMode() first and this one will throw. The question is (and the reason why both methods have the `REVIEW` comment) - should GetTransferMode() throw or should it assume Text if the feature is missing (I am leaning towards using text). If it does not throw (i.e. we allow to start the connection without setting the ITransferModeFeature) this one should just add the feature.